CMakeLists.txt

```diff
@@ -0,0 +1,186 @@
+set( ECBUILD_DEFAULT_BUILD_TYPE Release )
+
+# Enable MPI
+set( ENABLE_MPI ON CACHE BOOL "Compile with MPI")
+
+# Initialize
+ecbuild_bundle_initialize()
+ecbuild_bundle( PROJECT jedicmake GIT "https://github.com/jcsda-internal/jedi-cmake.git" BRANCH develop UPDATE )
+include( jedicmake/cmake/Functions/git_functions.cmake  )
+
+# ECMWF libs
+# ----------
+option("BUNDLE_SKIP_ECKIT" "Don't build eckit" "ON" ) # Skip eckit build unless user passes -DBUNDLE_SKIP_ECKIT=OFF
+option("BUNDLE_SKIP_FCKIT" "Don't build fckit" "ON") # Skip fckit build unless user passes -DBUNDLE_SKIP_FCKIT=OFF
+option("BUNDLE_SKIP_ATLAS" "Don't build atlas" "ON") # Skip atlas build unless user passes -DBUNDLE_SKIP_ATLAS=OFF
+
+ecbuild_bundle( PROJECT eckit GIT "https://github.com/ecmwf/eckit.git" TAG 1.16.0 )
+ecbuild_bundle( PROJECT fckit GIT "https://github.com/ecmwf/fckit.git" TAG 0.9.2 )
+ecbuild_bundle( PROJECT atlas GIT "https://github.com/ecmwf/atlas.git" TAG 0.24.1 )
+
+# External (required) observation operators
+# ------------------------------
+option("BUNDLE_SKIP_CRTM" "Don't build CRTM" "OFF") # Build crtm unless user passes -DBUNDLE_SKIP_CRTM=ON
+ecbuild_bundle( PROJECT crtm GIT "https://github.com/jcsda-internal/crtm.git" BRANCH release/crtm_jedi UPDATE )
+
+# Core JEDI repositories
+# ----------------------
+ecbuild_bundle( PROJECT oops  GIT "https://github.com/jcsda-internal/oops.git"  BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT saber GIT "https://github.com/jcsda-internal/saber.git" BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT ioda  GIT "https://github.com/jcsda-internal/ioda.git"  BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT ufo   GIT "https://github.com/jcsda-internal/ufo.git"   BRANCH develop UPDATE )
+
+# Options for building with certain models
+# ----------------------------------------
+set(UFS_APP "ATM" CACHE STRING "Choose the UFS Application")
+set_property(CACHE UFS_APP PROPERTY STRINGS "ATM" "NG-GODAS" "S2S")
+if(NOT UFS_APP MATCHES "^(ATM|NG-GODAS|S2S)$")
+  ecbuild_error("Valid options for -DUFS_APP are ATM|NG-GODAS|S2S")
+endif()
+ecbuild_info("Building with UFS application: ${UFS_APP}")
+
+# fv3-jedi linear model
+# ---------------------
+ecbuild_bundle( PROJECT fv3-jedi-lm GIT "https://github.com/jcsda-internal/fv3-jedi-linearmodel.git" BRANCH feature/ufs UPDATE )
+
+include_directories(${DEPEND_LIB_ROOT}/include)
+link_directories(${DEPEND_LIB_ROOT}/lib)
+
+include( ExternalProject )
+
+ExternalProject_Add(ufs-weather-model
+  PREFIX ${CMAKE_CURRENT_BINARY_DIR}/ufs-weather-model
+  GIT_REPOSITORY https://github.com/mark-a-potts/ufs-weather-model.git
+  GIT_SUBMODULES_RECURSE TRUE
+  GIT_TAG develop
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ufs-weather-model
+  INSTALL_DIR ${DEPEND_LIB_ROOT}
+  CMAKE_ARGS ${UFS_WEATHER_MODEL_ARGS}
+  CMAKE_ARGS -DCCPP_SUITES=FV3_GFS_v16,FV3_GFS_v16_coupled,FV3_GFS_v15p2 -DREPRO=on -DOPENMP=ON -DINLINE_POST=OFF -DMULTI_GASES=OFF -DMPI=ON -DAPP=ATM -DCMAKE_INSTALL_PREFIX=${DEPEND_LIB_ROOT}
```

Comment:
Maybe `-DAPP=${UFS_APP}`

Reply:
good catch.  I failed to add that in the `CMAKE_ARGS`.

Reply:
Done in 3dbd3df

Reply:
`CCPP_SUITES` will need to be updated based on the `UFS_APP` as well.  But that can be handled in another PR later.